common/blobstore/filestore/client.go

```diff
@@ -35,121 +37,179 @@ const (
 	metadataFilename = "metadata"
 )
 
+var (
+	// ErrCheckBucketExists could not verify that bucket directory exists
+	ErrCheckBucketExists = &shared.BadRequestError{Message: "could not verify that bucket directory exists"}
+	// ErrWriteFile could not write file
+	ErrWriteFile = &shared.BlobstoreNonRetryableError{Message: "could not write file"}
+	// ErrReadFile could not read file
+	ErrReadFile = &shared.BlobstoreNonRetryableError{Message: "could not read file"}
+	// ErrCheckFileExists could not check if file exists
+	ErrCheckFileExists = &shared.BlobstoreNonRetryableError{Message: "could not check if file exists"}
+	// ErrDeleteFile could not delete file
+	ErrDeleteFile = &shared.BlobstoreNonRetryableError{Message: "could not delete file"}
+	// ErrListFiles could not list files
+	ErrListFiles = &shared.BlobstoreNonRetryableError{Message: "could not list files"}
+	// ErrConstructKey could not construct key
+	ErrConstructKey = &shared.BlobstoreNonRetryableError{Message: "could not construct key"}
+	// ErrMetadataFileNotExists metadata file not exists
+	ErrMetadataFileNotExists = &shared.BlobstoreNonRetryableError{Message: "metadata file not exists"}
+	// ErrBucketConfigDeserialization bucket config could not be deserialized
+	ErrBucketConfigDeserialization = &shared.BlobstoreNonRetryableError{Message: "bucket config could not be deserialized"}
+)
+
 type client struct {
 	sync.Mutex
 	storeDirectory string
+	logger         bark.Logger
 }
 
 // NewClient returns a new Client backed by file system
-func NewClient(cfg *Config) (blobstore.Client, error) {
+func NewClient(cfg *Config, logger bark.Logger) (blobstore.Client, error) {
+	logger = logger.WithField("storeDirectory", cfg.StoreDirectory)
 	if err := cfg.Validate(); err != nil {
+		logger.WithError(err).Error("failed to validate config")
 		return nil, err
 	}
 	if err := setupDirectories(cfg); err != nil {
+		logger.WithError(err).Error("failed to setup directories")
 		return nil, err
 	}
 	if err := writeMetadataFiles(cfg); err != nil {
+		logger.WithError(err).Error("failed to write metadata files")
 		return nil, err
 	}
 	return &client{
 		storeDirectory: cfg.StoreDirectory,
+		logger:         logger,
 	}, nil
 }
 
 func (c *client) Upload(_ context.Context, bucket string, key blob.Key, blob *blob.Blob) error {
 	c.Lock()
 	defer c.Unlock()
 
-	exists, err := directoryExists(bucketDirectory(c.storeDirectory, bucket))
-	if err != nil {
-		return err
-	}
-	if !exists {
+	bd := bucketDirectory(c.storeDirectory, bucket)
+	if exists, err := directoryExists(bd); err != nil {
+		c.logger.Errorf("Upload failed. Error: %v. Bucket: %v", err, bucket)
```

Comment:
Do not use 'Errorf', this makes log searching very hard.  Use tags for dynamic fields like Error and Bucket.